ccm_web/client/src/utils/canvasSectionNameValidator.ts

```diff
@@ -37,22 +37,9 @@ export class CanvasCoursesSectionNameValidator {
     this.course = course
   }
 
-  private async getSections (): Promise<CanvasCourseSection[]> {
-    let sections: CanvasCourseSection[] = []
-    await getCourseSections(this.course.id).then(s => {
-      sections = s
-    }).catch(error => {
-      throw new Error(error)
-    })
-    return sections
-  }
-
   public async validateSectionName (newName: string): Promise<ICanvasSectionNameInvalidError[]> {
     const errors: ICanvasSectionNameInvalidError[] = []
-    let sections: CanvasCourseSection[] = []
-    await this.getSections().then(s => {
-      sections = s
-    })
+    const sections = await getCourseSections(this.course.id)
```

Comment:
@chrisrrowland, maybe I missed something but `getSections` didn't really seem to be doing anything that `await doTask` wouldn't do automatically. Actually, because error is passed through the `Error()` constructor, we were loosing the subclass typing, including `UnauthorizedError`, which I need for `hasUnauthorized`. Let me know if you have any thoughts.

Reply:
Yeah.. I agree we should fix this so that all API call redirect call handled at `handleError()` block

Reply:
I think this change isn't that relevant any more, because the redirect would be handled by the first `getCourseSections`. Still, I think I'm going to keep this change, as it'll make non-`UnauthorizedError` subclasses available to the error alerts. It's also more concise 😀